lib/auth/clusterconfig/clusterconfigv1/service.go

```diff
@@ -245,3 +250,180 @@ func (s Service) ResetAuthPreference(ctx context.Context, req *clusterconfigpb.R
 
 	return authPrefV2, nil
 }
+
+// GetClusterNetworkingConfig returns the locally cached networking configuration.
+func (s *Service) GetClusterNetworkingConfig(ctx context.Context, _ *clusterconfigpb.GetClusterNetworkingConfigRequest) (*types.ClusterNetworkingConfigV2, error) {
+	authzCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if err := authzCtx.CheckAccessToKind(types.KindClusterNetworkingConfig, types.VerbRead); err != nil {
+		if err2 := authzCtx.CheckAccessToKind(types.KindClusterConfig, types.VerbRead); err2 != nil {
+			return nil, trace.Wrap(err)
+		}
+	}
+
+	netConfig, err := s.cache.GetClusterNetworkingConfig(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	netConfigV2, ok := netConfig.(*types.ClusterNetworkingConfigV2)
+	if !ok {
+		return nil, trace.BadParameter("unexpected type %T", netConfig)
+	}
+	return netConfigV2, nil
+}
+
+// CreateClusterNetworkingConfig creates a new cluster networking configuration if one does not exist.
+// This is an internal API and is not exposed via [clusterconfigv1.ClusterConfigServiceServer]. It
+// is only meant to be called directly from the first time an Auth instance is started.
+func (s *Service) CreateClusterNetworkingConfig(ctx context.Context, cfg types.ClusterNetworkingConfig) (types.ClusterNetworkingConfig, error) {
+	authzCtx, err := s.authorizer.Authorize(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	if !authz.HasBuiltinRole(*authzCtx, string(types.RoleAuth)) {
+		return nil, trace.AccessDenied("this request can be only executed by an auth server")
+	}
+
+	tst, err := cfg.GetTunnelStrategyType()
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	if tst == types.ProxyPeering && modules.GetModules().BuildType() != modules.BuildEnterprise {
+		return nil, trace.AccessDenied("proxy peering is an enterprise-only feature")
+	}
+
+	created, err := s.backend.CreateClusterNetworkingConfig(ctx, cfg)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	cfgV2, ok := created.(*types.ClusterNetworkingConfigV2)
+	if !ok {
+		return nil, trace.Wrap(trace.BadParameter("unexpected type %T", created))
```

Comment:
nit: type error message here and in a few other places is a bit vague.

```suggestion
		return nil, trace.Wrap(trace.BadParameter("unexpected cluster networking config type %T (expected %T)", created, cfgV2))
```